app/src/main/java/org/fossasia/pslab/fragment/LALogicLinesFragment.java

```diff
@@ -477,8 +531,19 @@ private void singleChannelOtherEdges(double[] xData, double[] yData) {
         for (int i = 0; i < xData.length; i++) {
             int xaxis = (int) xData[i];
             int yaxis = (int) yData[i];
-            tempInput.add(new Entry(xaxis, yaxis));
+            tempInput.add(new Entry(xaxis, yaxis + 2 * currentChannel));
         }
+
+        // Adding the entry for current channel
+        LineDataSet lineDataSet = new LineDataSet(tempInput, channelNames.get(currentChannel));
```

Comment:
@harsh-2711 I observed these set of lines in three places. Please create a method with parameters like linedataset and call that method in these three places.